power_grid_model_c/power_grid_model/include/power_grid_model/math_solver/necesary_observability_check.hpp

```diff
@@ -0,0 +1,69 @@
+// SPDX-FileCopyrightText: Contributors to the Power Grid Model project <powergridmodel@lfenergy.org>
+//
+// SPDX-License-Identifier: MPL-2.0
+
+#pragma once
+
+#include "measured_values.hpp"
+
+#include "../common/exception.hpp"
+
+namespace power_grid_model::math_solver {
+
+template <symmetry_tag sym> class NecesaryObservabilityCheck {
+
+  public:
+    // construct
+    NecesaryObservabilityCheck(MeasuredValues<sym> const& measured_values,
```

Comment:
is there a necessity to make it symmetric? why not just

```suggestion
class NecesaryObservabilityCheck {

  public:
    // construct
    template <symmetry_tag sym>
    NecesaryObservabilityCheck(MeasuredValues<sym> const& measured_values,
```